### PR TITLE
feat: add "avoid jaas" validator to existing resources

### PR DESCRIPTION
**Pull request by kian99**
## Description

This PR adds "Avoid JAAS" validator to all resources that cannot be used with JAAS. Specifically:
- Ensure certain resources cannot be used with JAAS and add tests for this.
- Tweak the error messaging based on previous feedback in [this discussion](https://github.com/juju/terraform-provider-juju/pull/541#discussion_r1715287183).

Fixes: [CSS-10238](https://warthogs.atlassian.net/browse/CSS-10238)

## Type of change

- Logic changes in resources (additional validation for JAAS)

[CSS-10238]: https://warthogs.atlassian.net/browse/CSS-10238?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: internal/juju/client.go
Comment (hmlanigan) [inline internal/juju/client.go:135]: As you changed this also in a newly approved pr, please rebase once it lands.

File: internal/provider/validator_require_jaas.go
Comment (hmlanigan) [inline internal/provider/validator_require_jaas.go:24]: todo: add this change to the commit message.

File: internal/provider/resource_access_model.go
Comment (hmlanigan) [inline internal/provider/resource_access_model.go:56]: todo:
missing from file, there is a list of these at line 29.
```
var _ resource.ResourceWithConfigValidators = &accessModelResource{}
```

File: internal/provider/resource_user.go
Comment (hmlanigan) [inline internal/provider/resource_user.go:55]: todo:
missing from file, there is a list of these at line 24.
```
var _ resource.ResourceWithConfigValidators = &userResource{}
```

File: internal/provider/resource_access_generic.go
Comment (hmlanigan) [inline internal/provider/resource_access_generic.go:75]: todo:
missing from resource_access_jaas_model, there is a list of these at line 20.
```
var _ resource.ResourceWithConfigValidators = &jaasAccessModelResource{}
```

File: internal/juju/client.go
Comment (kian99) [reply internal/juju/client.go]: Done.

**Review by hmlanigan — APPROVED** [review]

**Pull request merged**
